README
Map-Tube-NYC

The README  is  used  to  introduce the module and provide instructions on how to
install the module, any machine dependencies it may have (for example C compilers
and installed libraries) and any other information that should be provided before
the module is installed.

A  README  file  is required for CPAN modules since CPAN extracts the README file
from a module distribution so that people browsing the archive can use it to  get
an  idea  of  the  module's  uses.  It  is usually a good idea to provide version
information here so that people can decide whether fixes for the module are worth
downloading.

INSTALLATION

To install this module, run the following commands:

	perl Makefile.PL
	make
	make test
	make install

SUPPORT AND DOCUMENTATION

After installing,  you  can  find  documentation for this module with the perldoc
command.

    perldoc Map::Tube::NYC

You can also look for information at:

    RT, CPAN's request tracker (report bugs here)
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=Map-Tube-NYC

    AnnoCPAN, Annotated CPAN documentation
        http://annocpan.org/dist/Map-Tube-NYC

    CPAN Ratings
        http://cpanratings.perl.org/d/Map-Tube-NYC

    Search CPAN
        http://search.cpan.org/dist/Map-Tube-NYC/

LICENSE AND COPYRIGHT

Copyright (C) 2014 - 2015 Mohammad S Anwar

This program is free software; you can redistribute it and/or modify it under the
terms of the the Artistic License (2.0).You may obtain a copy of the full license
at:

L<http://www.perlfoundation.org/artistic_license_2_0>

Any  use,  modification, and distribution of the Standard or Modified Versions is
governed by this Artistic License.By using, modifying / distributing the Package,
you accept this license. Do not use, modify, or distribute the Package, if you do
not accept this license.

If your Modified Version has been derived from a Modified Version made by someone
other than you, you're nevertheless required to ensure that your Modified Version
complies with the requirements of this license.

This  license  does  not  grant you the right to use any trademark, service mark,
tradename, or logo of the Copyright Holder.

This license includes the non-exclusive, worldwide, free-of-charge patent license
to  make,  have made, use, offer to sell, sell, import and otherwise transfer the
Package with respect to any patent claims licensable by the Copyright Holder that
are  necessarily  infringed  by the  Package.  If you institute patent litigation
(including  a  cross-claim or  counterclaim)  against any party alleging that the
Package constitutes direct or contributory patent infringement,then this Artistic
License to you shall terminate on the date that such litigation is filed.

Disclaimer  of  Warranty:  THE  PACKAGE  IS  PROVIDED BY THE COPYRIGHT HOLDER AND
CONTRIBUTORS  "AS IS'  AND WITHOUT ANY EXPRESS OR IMPLIED WARRANTIES. THE IMPLIED
WARRANTIES    OF     MERCHANTABILITY,   FITNESS   FOR   A PARTICULAR  PURPOSE, OR
NON-INFRINGEMENT ARE DISCLAIMED TO THE EXTENT PERMITTED BY YOUR LOCAL LAW. UNLESS
REQUIRED BY LAW, NO COPYRIGHT HOLDER OR CONTRIBUTOR WILL BE LIABLE FOR ANY DIRECT,
INDIRECT, INCIDENTAL, OR CONSEQUENTIAL DAMAGES ARISING IN ANY WAY OUT OF THE USE
OF THE PACKAGE,EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.


Changes
Revision history for Map::Tube::NYC

0.67  Mon Jun 24 17:50:00 2019
      - Updated dependency on Map::Tube v3.63.

0.66  Fri May 17 10:00:00 2019
      - Updated dependency on Map::Tube v3.62.

0.65  Wed Apr 24 14:30:00 2019
      - Updated dependency on Map::Tube v3.61.

0.64  Mon Mar 04 16:45:00 2019
      - Updated dependency on Map::Tube v3.60.

0.63  Tue Feb 12 15:00:00 2019
      - Updated dependency on Map::Tube v3.59.

0.62  Fri Dec 28 03:40:00 2018
      - Updated dependency on Map::Tube v3.58.

0.61  Sat Oct 13 08:20:00 2018
      - Updated dependency on Map::Tube v3.57.

0.60  Thu Sep 13 18:10:00 2018
      - Updated dependency on Map::Tube v3.56.

0.59  Sat Aug 18 07:40:00 2018
      - Updated dependency on Map::Tube v3.55.

0.58  Wed Jul 11 12:00:00 2018
      - Used File::ShareDir v1.114 instead of File::Share.
      - Explicitly added configure prereq File::ShareDir::Install v0.13.

0.57  Tue Jul 03 12:25:00 2018
      - Updated dependency on Map::Tube v3.53.

0.56  Wed Jun 20 18:10:00 2018
      - Used new_ok() in test script.

0.55  Fri Jun 08 06:00:00 2018
      - Updated dependency on Map::Tube v3.51.

0.54  Mon May 21 14:00:00 2018
      - Tidied up code in the pod.

0.53  Fri May 11 07:35:00 2018
      - Updated dependency on Test::Map::Tube v0.44.

0.52  Thu May 03 14:30:00 2018
      - Updated dependency on Map::Tube v3.50.

0.51  Wed Apr 11 14:00:00 2018
      - Updated dependency on Map::Tube v3.49.

0.50  Fri Mar 23 18:05:00 2018
      - Updated dependency on Map::Tube v3.45.

0.49  Sat Feb 03 22:10:00 2018
      - Updated dependency on Map::Tube v3.44.

0.48  Fri Dec 29 10:40:00 2017
      - Updated dependency on Map::Tube v3.43.

0.47  Mon Dec 04 11:45:00 2017
      - Updated dependency on Map::Tube v3.42.
      - Replaced namespace:clean with namespace::autoclean.

0.46  Mon Nov 20 01:35:00 2017
      - Updated dependency on Map::Tube v3.40.

0.45  Fri Nov 10 17:05:00 2017
      - Updated dependency on Map::Tube v3.37.
      - Upgraded to use Test::Map::Tube v0.35.

0.44  Sat Oct 28 09:45:00 2017
      - Upgraded to use Map::Tube v3.36.

0.43  Mon Oct 16 09:00:00 2017
      - Upgraded to use Map::Tube v3.35.

0.42  Wed Aug 23 12:00:00 2017
      - Upgraded to use Map::Tube v3.34.

0.41  Mon Jul 17 11:00:00 2017
      - Upgraded to use Map::Tube v3.32.

0.40  Sat Jun 03 00:00:00 2017
      - Upgraded to use Map::Tube v3.30.

0.39  Sun May 21 09:25:00 2017
      - Upgraded to use Map::Tube v3.29.

0.38  Sun Mar 19 19:55:00 2017
      - Upgraded to use Test::Map::Tube v0.25.

0.37  Sat Mar 11 11:10:00 2017
      - Upgraded to use Map::Tube v3.28.

0.36  Tue Oct 25 14:25:00 2016
      - Upgraded to use Map::Tube v3.21.
      - Upgraded to use Test::Map::Tube v0.17.

0.35  Sat Aug 20 12:45:00 2016
      - Upgraded to use Map::Tube v3.18.

0.34  Tue Jun 28 10:20:00 2016
      - Upgraded to use Map::Tube v3.15.
      - Consolidated routes testing functions using method ok_map_routes().

0.33  Wed Apr 13 12:10:00 2016
      - Added test script for Line IRT Lexington Avenue.
      - Fixed typo in the pod document.
      - Updated map data w.r.t. "Spring Street" station.

0.32  Tue Apr 05 10:40:00 2016
      - Added package Map::Tube::NYC::Line::IRTLexingtonAvenue
      - Added map image for IRT Lexington Avenue line.
      - Updated NYC Submap map image.
      - Updated effected other line map image.
      - Updated unit test t/00-load.t
      - Updated map data w.r.t. "Grand Central" station.

0.31  Fri Apr 01 16:15:00 2016
      - Fixed typo in the package name.
      - Updated test scripts.
      - Updated copyright year.

0.30  Tue Mar 29 15:25:00 2016
      - Added map data for line 'IRT Lexington Avenue'.

0.29  Fri Mar 11 11:05:00 2016
      - Added map for BMT Nassau Street line.
      - Added unit test for BMT Nassau Street line.

0.28  Sun Feb 28 09:45:00 2016
      - Updated NYC Subway map.
      - Added map for BMT Canarsie line.
      - Added unit test for BMT Canarsie line.

0.27  Fri Feb 26 18:10:00 2016
      - Added line BMT Canarsie.

0.26  Fri Nov 06 11:15:00 2015
      - Fixed the error reported by CPANT Testers.
      - Upgraded to use Map::Tube v3.12.

0.25  Fri Sep 25 10:20:00 2015
      - Upgraded to use Map::Tube v3.01.

0.24  Sun Jul 26 12:00:00 2015
      - Upgraded to use Map::Tube v3.00.

0.23  Mon Mar 30 09:00:00 2015
      - Fixed the station '23rd Street' data in the xml file.

0.22  Thu Feb 26 16:30:00 2015
      - Added package to list subway stations for "IRT Flushing Line".
      - Added unit test for the "IRT Flushing Line".
      - Added map for the "IRT Flushing Line".
      - Updated NYC Subway map.

0.21  Thu Feb 26 11:25:00 2015
      - Added File::ShareDir::Install as configure requires in the Makefile.PL file.

0.20  Wed Feb 25 17:40:00 2015
      - Updated description for method get_shortest_route() in the pod document.

0.19  Wed Feb 25 17:15:00 2015
      - Tidied up the pod document.

0.18  Wed Feb 25 15:25:00 2015
      - Upgraded to use Map::Tube v2.87.
      - Added package to list subway stations for "IND Crosstown Line".
      - Added unit test for the "IND Crosstown Line".
      - Added map for the "IND Crosstown Line".
      - Updated NYC Subway map.

0.17  Mon Feb 16 17:50:00 2015
      - Upgraded to use Map::Tube v2.83.
      - Updated visual settings in the pod document for IND Sixth Avenue line map.
      - Updated sample code in the pod document.

0.16  Mon Feb 16 15:25:00 2015
      - Updated link information for the station "West Fourth Street" in the xml data file.
      - Updated station "Rockefeller Center" in the xml data file.
      - Added unit test for the IND Sixth Avenue line.
      - Added NYC Subway map.
      - Added IND Sixth Avenue line map.
      - Updated IND Eighth Avenue line map.

0.15  Mon Feb 16 12:55:00 2015
      - Added package to list subway stations for "IND Sixth Avenue".
      - Updated xml data and added stations for the above line.

Makefile.PL
#!/usr/bin/perl

use 5.006;
use strict;
use warnings FATAL => 'all';
use ExtUtils::MakeMaker;
use File::ShareDir::Install;

install_share dist => 'share';

WriteMakefile(
    NAME               => 'Map::Tube::NYC',
    AUTHOR             => q{Mohammad S Anwar <mohammad.anwar@yahoo.com>},
    VERSION_FROM       => 'lib/Map/Tube/NYC.pm',
    ABSTRACT_FROM      => 'lib/Map/Tube/NYC.pm',
    LICENSE            => 'artistic_2',
    MIN_PERL_VERSION   => 5.006,
    CONFIGURE_REQUIRES => {
        'ExtUtils::MakeMaker'     => 0,
        'File::ShareDir::Install' => '0.13',
    },
    BUILD_REQUIRES   => {
        'Test::More' => 0,
    },
    PREREQ_PM => {
        'Moo'                  => '2.000000',
        'namespace::autoclean' => '0.28',
        'File::ShareDir'       => '1.114',
        'Map::Tube'            => '3.63',
    },
    dist  => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
    clean => { FILES => 'Map-Tube-NYC-*' },
    (eval { ExtUtils::MakeMaker->VERSION(6.46) } ? (META_MERGE => {
        'meta-spec' => { version => 2 },
        provides    => {
            'Map::Tube::NYC'                           => { file => 'lib/Map/Tube/NYC.pm'                        , version => '0.67' },
            'Map::Tube::NYC::Line::INDSixthAvenue'     => { file => 'lib/Map/Tube/NYC/Line/INDSixthAvenue.pm'    , version => '0.67' },
            'Map::Tube::NYC::Line::INDEighthAvenue'    => { file => 'lib/Map/Tube/NYC/Line/INDEighthAvenue.pm'   , version => '0.67' },
            'Map::Tube::NYC::Line::INDCrosstown'       => { file => 'lib/Map/Tube/NYC/Line/INDCrosstown.pm'      , version => '0.67' },
            'Map::Tube::NYC::Line::IRTFlushing'        => { file => 'lib/Map/Tube/NYC/Line/IRTFlushing.pm'       , version => '0.67' },
            'Map::Tube::NYC::Line::BMTCanarsie'        => { file => 'lib/Map/Tube/NYC/Line/BMTCanarsie.pm'       , version => '0.67' },
            'Map::Tube::NYC::Line::BMTNassauStreet'    => { file => 'lib/Map/Tube/NYC/Line/BMTNassauStreet.pm'   , version => '0.67' },
            'Map::Tube::NYC::Line::IRTLexingtonAvenue' => { file => 'lib/Map/Tube/NYC/Line/IRTLexingtonAvenue.pm', version => '0.67' },
        },
        resources => {
            repository => {
                type => 'git',
                url  => 'https://github.com/manwar/Map-Tube-NYC.git',
                web  => 'https://github.com/manwar/Map-Tube-NYC',
            },
        }})
     : ()
    ),
);

package MY;
use File::ShareDir::Install 'postamble';
